admin/ops/setup_dev_bucket.py

```diff
@@ -0,0 +1,342 @@
+import boto3
+import importlib
+import os
+from pathlib import Path
+import typer
+from typing import Literal
+
+PROD_RECIPES_BUCKET = "edm-recipes"
+PROD_PUBLISHING_BUCKET = "edm-publishing"
+os.environ["RECIPES_BUCKET"] = PROD_RECIPES_BUCKET
+os.environ["PUBLISHING_BUCKET"] = PROD_PUBLISHING_BUCKET
+
+from dcpy import configuration
+from dcpy.models.connectors.edm.recipes import DatasetKey
+from dcpy.models.connectors.edm.publishing import (
+    ProductKey,
+    BuildKey,
+    DraftKey,
+    PublishKey,
+)
+from dcpy.utils.s3 import get_subfolders
+from dcpy.connectors.edm import recipes, publishing
+from dcpy.lifecycle.builds import plan
+
+
+ROOT_PATH = Path(__file__).parent.parent.parent
+DEV_BUCKET_PREFIX = "de-dev-"
+ACL: Literal["private"] = "private"
+PRODUCTS = [
+    "db-cbbr",
+    "db-colp",
+    "db-cpdb",
+    "db-developments",
+    "db-facilities",
+    "db-green_fast_track",
+    "db-template",
+    "db-pluto",
+    "db-zoningtaxlots",
+]
+
+
+def copy_folder_to_dev_bucket(source_bucket, target_bucket, source_path, target_path):
+    """
+    Thin wrapper over s3.copy_folder
+    Ensures that bucket being copied to is a dev bucket
+    """
+    from dcpy.utils import s3
+
+    assert target_bucket.startswith(DEV_BUCKET_PREFIX)
+    s3.copy_folder(
+        bucket=source_bucket,
+        target_bucket=target_bucket,
+        source_path=source_path,
+        target_path=target_path,
+        acl=ACL,
+    )
+
+
+def resolve_latest_recipe(
+    target_bucket: str,
+    ds: str,
+):
+    """
+    Within a dev bucket, resolves/copies the dataset currently in "latest"
+    into its versioned folder under "datasets/{dataset_id}"
+    """
+    os.environ["RECIPES_BUCKET"] = target_bucket
+    importlib.reload(configuration)
```

Comment:
What's the reason for reloading these? This does make me think that we need a more comprehensive way to load env vars. And maybe that module can have an `ensure_dev` function, to error out if detects prod variables. Not something we need to address now, but something to think about.